modules/agent/pkg/deployer/monitor.go

```diff
@@ -20,12 +31,144 @@ type DeploymentStatus struct {
 	ModifiedStatus []fleet.ModifiedStatus `json:"modifiedStatus,omitempty"`
 }
 
-func (m *Manager) plan(bd *fleet.BundleDeployment, ns string, objs ...runtime.Object) (apply.Plan, error) {
+func (m *Manager) plan(ctx context.Context, bd *fleet.BundleDeployment, ns string, objs ...runtime.Object) (apply.Plan, error) {
 	a, err := m.getApply(bd, ns)
 	if err != nil {
 		return apply.Plan{}, err
 	}
-	return a.DryRun(objs...)
+	plan, err := a.DryRun(objs...)
+	if err != nil {
+		return apply.Plan{}, err
+	}
+
+	result := objectset.ObjectByGVK{}
+	for _, obj := range objs {
+		result.Add(obj)
+	}
+
+	norms, err := m.normalizers(bd)
+	if err != nil {
+		return plan, err
+	}
+
+	for gvk, objs := range plan.Update {
```

Comment:
If you bump to the latest wrangler the sharedclientfactory (which you can obtain from the sharedcachefactory) there is a IsNamespaced method: https://github.com/rancher/lasso/blob/master/pkg/client/sharedclientfactory.go#L30

Reply:
:+1: